tools/generate-llms-files.py

```
@@ -402,14 +404,18 @@ def main():
-        resolved_xml_path = os.path.abspath(xml_path)
-        if not is_safe_path(resolved_xml_path):
+        resolved_xml_path = os.path.realpath(os.path.abspath(xml_path))
+        if os.path.commonpath([base_dir_abs, resolved_xml_path]) != base_dir_abs:
             print(f"Error: Path traversal blocked on XML output file '{xml_path}'.", file=sys.stderr)
             sys.exit(1)
 
         xml_content = generate_xml_context(parsed, args.base_dir)
         xml_success = False
         tmp_xml_path = resolved_xml_path + ".tmp"
+        if os.path.commonpath([base_dir_abs, tmp_xml_path]) != base_dir_abs:
+            print(f"Error: Path traversal blocked on temporary XML file.", file=sys.stderr)
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)